### ISSUE TITLE
Possible memory leak when recreating component with sizeMe

### ISSUE THREAD

**Issue by rodinhart** [ctrlplusb/react-sizeme#160]
I have an application that creates and mounts a sized component multiple times, something like:

```js
return flag ? <OtherComp /> : sizeMe({ monitorWidth: true, monitorHeight: true })(SizedComp)
```
 So depending on user actions, flag will toggle multiple times.

The following function caches instances of element-resize-detector:

```js
function resizeDetector(strategy = 'scroll') {
  if (!instances[strategy]) {
    instances[strategy] = createResizeDetector({
      strategy,
    })
  }

  return instances[strategy]
}
```
(from https://github.com/ctrlplusb/react-sizeme/blob/103dafbcc315b488d9aca2e8a7a9510e5669a382/src/resize-detector.js#L7)

Now, it seems that although there is only one entry in `instances` (with key `scroll`), it is holding on to multiple instances of `SizedComp` in our example. I can see multiple instance in a heap snapshot in chrome.
In other words, if I remove the caching and just return a new `createResizeDetector` everytime, or put `instances` in `global` and clear out the object when `flag` toggles, the instances of `SizedComp` disappear.

Is this the right way to use it, and if so, how can it be fixed?

**Comment by ctrlplusb** [comment]
Howdy;

I recently made some changes to the way events are uninstalled. Could possibly have been a previous issue. Anyways, I ran the profiler against a throw up app that continuously adds and removes resize instances. It looks like GC is behaving correctly.

<img width="828" alt="Screenshot 2019-05-01 14 22 14" src="https://user-images.githubusercontent.com/12164768/57018830-b8fd4180-6c1c-11e9-8c3a-fb8ee3d2088e.png">
